src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraService.scala

```diff
@@ -42,17 +43,21 @@ abstract class AgoraService(permissionsDataSource: PermissionsDataSource) extend
             val entityTypes = AgoraEntityType.byPath(path)
 
             get {
-              val queryAttempt = agoraBusiness.findSingle(targetEntity, entityTypes, username)
-
-              onComplete(queryAttempt) {
-                case Success(foundEntity) =>
-                  (onlyPayload, payloadAsObject) match {
-                    case (true, true) => complete(BadRequest, "onlyPayload, payloadAsObject cannot be used together")
-                    case (true, false) => complete(foundEntity.payload)
-                    case (false, true) => complete(foundEntity.withDeserializedPayload)
-                    case _ => complete(foundEntity)
+              Try(agoraBusiness.findSingle(targetEntity, entityTypes, username)) match {
+                case Success(queryAttempt) =>
+                  onComplete(queryAttempt) {
+                    case Success(foundEntity) =>
+                      (onlyPayload, payloadAsObject) match {
+                        case (true, true) => complete(BadRequest, "onlyPayload, payloadAsObject cannot be used together")
```

Comment:
this `(true, true)` case could be moved up higher, before we hit the databases for anything - fail fast!